### ISSUE TITLE
GUI Display Issue Unifi Wifi 

### ISSUE THREAD

**Issue by TyShawn76** [librenms/librenms#5766]
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

GitHub issues is for feature requests or bugs, please do not post issues asking for help or how to do X, Y or Z. 
You can use our irc channel ##librenms on freenode to ask questions or our [community site](https://community.librenms.org).

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [x] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [x] Please include all of the information between the `====================================` section of `./validate.php` which you can run from the cli.
==========================================================
Component | Version
--------- | -------
LibreNMS  | 424ac80bff355ebcb503d1730928e06cdb70f7af
DB Schema | 159
PHP       | 5.6.29-0+deb8u1
MySQL     | 5.5.54-0+deb8u1
RRDTool   | 1.4.8
SNMP      | NET-SNMP 5.7.2.1
==========================================================
- [x] Unless your issue is for a WebUI fix or feature then please provide ALL info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
- [x] Please provide as much detail as possible.
- [x] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.
Multiple sections are off in the GUI most are over Chrome and Safari and one is only over Safari.

The following section is in the Unifi wireless area, and the text is cut off over both browsers.
![screen shot 2017-02-04 at 6 50 58 am](https://cloud.githubusercontent.com/assets/21324034/22618504/038c367e-eaac-11e6-8e87-ecccecbc257a.png)
![screen shot 2017-02-04 at 6 51 14 am](https://cloud.githubusercontent.com/assets/21324034/22618505/03920e0a-eaac-11e6-8fe4-11df0fcc2e8f.png)

**Comment by laf** [comment]
Can you try this patch please:

```
diff --git a/html/includes/graphs/device/ubnt_unifi_RadioCu_0.inc.php b/html/includes/graphs/device/ubnt_unifi_RadioCu_0.inc.php
index 3d0e246..3968db0 100644
--- a/html/includes/graphs/device/ubnt_unifi_RadioCu_0.inc.php
+++ b/html/includes/graphs/device/ubnt_unifi_RadioCu_0.inc.php
@@ -32,4 +32,4 @@ if (rrdtool_check_rrd_exists($rrd_filename)) {
     echo "file missing: $rrd_filename";
 }

-require 'includes/graphs/generic_multi.inc.php';
+require 'includes/graphs/generic_multi_line.inc.php';
diff --git a/html/includes/graphs/device/ubnt_unifi_RadioCu_1.inc.php b/html/includes/graphs/device/ubnt_unifi_RadioCu_1.inc.php
index 58532bc..5200a89 100644
--- a/html/includes/graphs/device/ubnt_unifi_RadioCu_1.inc.php
+++ b/html/includes/graphs/device/ubnt_unifi_RadioCu_1.inc.php
@@ -32,4 +32,4 @@ if (rrdtool_check_rrd_exists($rrd_filename)) {
     echo "file missing: $rrd_filename";
 }

-require 'includes/graphs/generic_multi.inc.php';
+require 'includes/graphs/generic_multi_line.inc.php';

```

**Comment by TyShawn76** [comment]
Sure how to I apply this patch?

**Comment by laf** [comment]
Copy the contents into a file called patch in /opt/librenms/ then run: `git apply patch`

**Comment by TyShawn76** [comment]
Patch seems to have resolved the issue.

**Comment by laf** [comment]
Tagging @jquagga

**Comment by jquagga** [comment]
Hi @laf / @TyShawn76,

So I looked at this and I think it's an issue with ```html/includes/graphs/generic_multi_line.inc.php``` as it's not only the Unifi graphs which show this behavior (at least on my installed).  Here's a screenshot of the netstat's snmp poller graph and it has the same "falling off the edge" problem.  

![netstats-fs8](https://cloud.githubusercontent.com/assets/5832772/22741205/58859fe4-ede0-11e6-89a0-cab919e1fae0.png)

I can offer a fix but it may be sub-optimal.  I looked at generic_multi_line (a different "generic" but one we know works) and it always sets ```$descr_len = 12```.  generic_multi_line (which both Unifi and netstats-snmp and a few other graphs use) has some fancier math going on and does different things for wide and thin graphs.  The line which causes the issue I believe is line 24 of generic_multi.inc.php:

```
$descr_len += round(($width - 250) / 8);
```

Prior to this, the length is set to 12.  This adds to it to optimize the width, but it makes the graph fall off the edge.  I'm afraid I don't understand this math well enough to determine what the optimal value is.  However, if you just comment the line out, the $descr_len becomes 12 and that works.  

After that line is commented out:
![wifi-fs8](https://cloud.githubusercontent.com/assets/5832772/22741419/39be6f2c-ede1-11e6-93c0-ca77607a3ded.png)

So on the one hand, this is sub-optimal as it doesn't push the right side of the graph all the way to the margin.  On the other hand, it matches the existing behavior of generic_multi_line.inc.php.  So it does look nice with the Wifi Clients graph (which uses the multi line generic).  

![wifi2-fs8](https://cloud.githubusercontent.com/assets/5832772/22741770/5753ced2-ede2-11e6-8ec0-cc7ae060a251.png)


Simplest "fix" patch:
```diff
diff --git a/html/includes/graphs/generic_multi.inc.php b/html/includes/graphs/generic_multi.inc.php
index e7b10b8d0..bbaef2ed5 100644
--- a/html/includes/graphs/generic_multi.inc.php
+++ b/html/includes/graphs/generic_multi.inc.php
@@ -21,7 +21,8 @@ if ($width > '500') {
     $descr_len = 24;
 } else {
     $descr_len  = 12;
-    $descr_len += round(($width - 250) / 8);
+    # FIXME - Presently overflows graph
+    #$descr_len += round(($width - 250) / 8);
 }
 
 if ($nototal) {
```
